src/server/gameEngine/gameEngine.ts

```diff
@@ -424,13 +426,16 @@ export const applyGameAction = ({
                 (card) => card.id === cardId
             );
 
-            if (matchingCard.cardType !== CardType.SPELL) {
+            if (matchingCard?.cardType !== CardType.SPELL) {
```

Comment:
the `?.` is the result of the fixing the bug mentioned in the spec file ^

In truth, the bug is highly unlikely to come up unless users are intentionally manipulating their redux stores via dev tools, but we want to prevent malicious users from crashing the whole node server with an unhandled error.

Basically, matchingCard could be potentially undefined because of the way that es6's `.find` method works.  We want to guard against an "can't access cardType of undefined" error and this puts safeguards on it (optional chaining is the name of this ES6 feature)